lens/private/list/flatten.scrbl

```diff
@@ -0,0 +1,63 @@
+#lang scribble/manual
+
+@(require "../doc-util/main.rkt")
+
+@title{Flattening and unflattening lists}
+
+@defthing[append*-lens lens?]{
+A lens that flattens a list one-level down when viewing, and restores
+the original structure when setting.
+@lens-unstable-examples[
+  (lens-view append*-lens '((a) (b c) () (d e f)))
+  (lens-set append*-lens '((a) (b c) () (d e f)) '(1 2 3 4 5 6))
+]}
+
+@defproc[(flatten/depth-lens [n exact-nonnegative-integer?]) lens?]{
+Creates a lens that flattens a list of depth @racket[n] when
+viewing, and restores the original structure when setting.
+
+Conceptually, @racket[append*-lens] is equivalent to
+@racket[(flatten/depth-lens 2)], since it takes a list of depth 2.
+@lens-unstable-examples[
+  (lens-view (flatten/depth-lens 1) '(a b c))
+  (lens-set (flatten/depth-lens 1) '(a b c) '(1 2 3 4 5 6))
+  (lens-view (flatten/depth-lens 2) '((a) (b c) () (d e f)))
+  (lens-set (flatten/depth-lens 2) '((a) (b c) () (d e f)) '(1 2 3 4 5 6))
+  (lens-view (flatten/depth-lens 3) '(((a) ()) (() (b) (c)) () ((d e) () (f))))
+  (lens-set (flatten/depth-lens 3) '(((a) ()) (() (b) (c)) () ((d e) () (f))) '(1 2 3 4 5 6))
+]}
+
+@defproc[(flatten/depth [n exact-nonnegative-integer?] [lst* any/c]) list?]{
+Flattens a list of depth @racket[n]. For depth @racket[n] = @racket[0],
+it returns @racket[(list lst*)]. For depth @racket[n] = @racket[1], it
+returns @racket[lst*], for depth @racket[2], it returns
+@racket[(append* lst*)], for depth @racket[3], it returns
+@racket[(append* (append* lst*))], and so on for higher depths.
+
+This is what @racket[flatten/depth-lens] uses for viewing.
```

Comment:
In addition to linking to the lens, `flatten/depth-lens` should document that it uses `flatten/depth` for viewing so they're both linked to each other.